cachi2/core/package_managers/gomod.py

```diff
@@ -1247,6 +1257,13 @@ def __init__(self, repo: git.Repo, commit: git.objects.commit.Commit):
         self._commit = commit
 
     @classmethod
+    @backoff.on_exception(
+        fast_constant_delay,
+        Exception,
+        jitter=None,  # use deterministic backoff, do not apply jitter
```

Comment:
I'm wondering why we concluded that jitter is bad (originally appeared first in commit 9d6f0603c migrating the logic from cachito).

Reply:
> Since we have effectively just one process I'd say we can safely rely on jitter occurring naturally.

This isn't true though for PaC services, where there are many concurrent pipelines running our tool as a standalone process each, potentially contacting the same service concurrently.

Reply:
> Initially I have implemented my own retry decorator (a much simpler one than that provided by backoff) only to realize that we have a dependency which I could reuse.

Well, you might as well propose it :) since backoff seems to have been abandoned, so that way, we could maintain our own simple implementation that would suit our needs just fine without additional cruft.

Reply:
> there are many concurrent pipelines running our tool

They still don't do this in sync. Usually artificial jitter is a small (on the order of tens percent maximum) deviation from certain value, and that's what we'll get when jobs are triggered randomly.